tests/unit/test_column_noise.py

```diff
@@ -131,9 +139,100 @@ def test_miswrite_ages():
     pass
 
 
-@pytest.mark.skip(reason="TODO")
-def test_miswrite_numerics():
-    pass
+def test_miswrite_numerics(string_series):
+    """
+    Validates that only numeric characters are noised in a series at a provided noise level.
+    """
+    config = get_configuration()
+    config.update(
+        {
+            "decennial_census": {
+                "street_number": {
+                    "numeric_miswriting": {
+                        "row_noise_level": 0.4,
+                        "token_noise_level": 0.5,
+                    },
+                },
+            },
+        }
+    )
+    config = config["decennial_census"]["street_number"]["numeric_miswriting"]
+    p_row_noise = config.row_noise_level
+    p_token_noise = config.token_noise_level
+    data = string_series
+    noised_data = _validate_seed_and_noise_data(
+        noise_type=NOISE_TYPES.NUMERIC_MISWRITING, column=data, config=config
+    )
+
+    # Get masks for helper groups, each string in categorical string purpose is to mimic possible string types
+    empty_str = data.str.len() == 0
+    unit_number = data.str.len() == 2
+    id_number = data.str.len() == 4
+    alt_str = data.str.len() == 6
+    income = data.str.len() == 9
+    date_of_birth = data.str.len() == 10
+    ssn = data.str.len() == 11
+    expected_noise = p_row_noise * p_token_noise * 0.9
+
+    # CHeck empty strings havent changed
+    assert (noised_data[empty_str] == "").all()
+
+    for i in range(2):  # "1A"
+        if i == 0:
+            assert np.isclose(
```

Comment:
Also add `assert noised_data[unit_number].str[i].isdigit().all()` to prove that these columns remain numbers. 

Same for all string types below.

Reply:
I've added both of these.  It does seem a bit redundant for the alphabetical characters since we check they haven't changed but it doesn't hurt to verify.